x-pack/plugins/observability/server/lib/rules/threshold/lib/create_custom_metrics_aggregations.ts

```diff
@@ -7,18 +7,18 @@
 
 import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
 import { isEmpty } from 'lodash';
-import { MetricExpressionCustomMetric } from '../../../../../common/threshold_rule/types';
+import { MetricExpressionMetrics } from '../../../../../common/threshold_rule/types';
 import { MetricsExplorerCustomMetric } from './metrics_explorer';
 
 const isMetricExpressionCustomMetric = (
-  subject: MetricsExplorerCustomMetric | MetricExpressionCustomMetric
-): subject is MetricExpressionCustomMetric => {
-  return (subject as MetricExpressionCustomMetric).aggType != null;
+  subject: MetricsExplorerCustomMetric | MetricExpressionMetrics
+): subject is MetricExpressionMetrics => {
+  return (subject as MetricExpressionMetrics).aggType != null;
```

Comment:
🍰 nit: If you want to avoid casting, you can use the [`in` operator](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing)

```suggestion
  return "aggType" in subject;
```